### ISSUE TITLE
Having a `.solhint.json` with custom rules breaks the Solidity compiler server

### ISSUE THREAD

**Issue by Jaime-Iglesias** [juanfranblanco/vscode-solidity#179]
![Screenshot from 2020-05-23 19-07-12](https://user-images.githubusercontent.com/22358726/82737400-d53cb680-9d28-11ea-93c3-6859d4fec40d.png)

I've noticed that my extension was not telling me about errors, like `unused parameters` and stuff like that, then I noticed the message about the Solidity server crashing.

At first  it was complaining about some custom rules that I have (They are for solhint `2.x`), then I removed the rules (but left the `.solhint.json`) then it complained about it being empty. After I completely removed the `.solhint.json` file it did not stop crashing.

- Temportal fix: Removing the `.solhint.json`

**Comment by juanfranblanco** [comment]
Thanks, I have upgraded to solhint 3.0 on the latest release, I did not experienced any issues on my side even using a JSON file... obviously not enough :(

**Comment by Jaime-Iglesias** [comment]
> Thanks, I have upgraded to solhint 3.0 on the latest release, I did not experienced any issues on my side even using a JSON file... obviously not enough :(

I will try installing the latest solhint version locally and use some custom rules that I already have for `3.0`, I'll  tell you how that goes :D

**Comment by juanfranblanco** [comment]
![image](https://user-images.githubusercontent.com/562371/82748148-93972480-9d97-11ea-82a5-32808609b1be.png)

**Comment by Jaime-Iglesias** [comment]
So, after some extra testing I've come to the following conclusions:

1. The `solhint` version is not the problem.
2. Having a `.solhint.json` is not the problem.
3. The problem is having custom rules.

- The error.
![Screenshot from 2020-05-24 12-03-51](https://user-images.githubusercontent.com/22358726/82752544-b258e380-9db6-11ea-9d8e-73265d0ae356.png)
- My `.solhint.json` file.
![Screenshot from 2020-05-24 12-04-43](https://user-images.githubusercontent.com/22358726/82752562-c56bb380-9db6-11ea-8dd7-8cb6a5d8bf22.png)
- My `package.json`
![Screenshot from 2020-05-24 12-05-19](https://user-images.githubusercontent.com/22358726/82752572-d9afb080-9db6-11ea-8267-9bbdb0bcc196.png)

**Comment by juanfranblanco** [comment]
Ah what you meant by custom rules was your plugins, not your specific rules. Yes that needs to be fixed by the solhint guys... the same as before..

**Comment by Jaime-Iglesias** [comment]
> Ah what you meant by custom rules was your plugins, not your specific rules. Yes that needs to be fixed by the solhint guys... the same as before..

Yeah, the problem is that it now crashes the `Solidity Compiler Server` which it didn't before :(

**Comment by paulrberg** [comment]
I'm wrestling with a similar issue now, after having recently upgraded to solhint@3.3.4. Some obvious errors are correctly highlighted:

<img width="518" alt="Capture d’écran 2021-04-27 à 16 40 01" src="https://user-images.githubusercontent.com/8782666/116252476-47f3b480-a778-11eb-8c44-1ec8a5423cb7.png">

But others, such as incorrect types, are not (`underlyingOut` should be an uint256):

<img width="518" alt="Capture d’écran 2021-04-27 à 16 42 35" src="https://user-images.githubusercontent.com/8782666/116252507-4e822c00-a778-11eb-8772-05841bf62b64.png">

This is with vscode-solidity@0.0.113. Deleting `.solhint.json` didn't fix this.

**Comment by juanfranblanco** [comment]
@paulrberg custom rules is not supported as it needs further work from solhint, mainly a way to inject those custom rules into the plugin. As those rules need to be loaded dynamically and found. Mainly in the user repo or something like that.

**Comment by paulrberg** [comment]
I don't think I am using custom rules. This is my [.solhint.json file](https://github.com/paulrberg/solidity-template/blob/8c266a343ac751b2b73f79783498fd35b5bc5f08/.solhint.json).

Also the type error above is not triggered by solhint, but the solidity compiler itself.

**Comment by juanfranblanco** [comment]
Yes you are using the "prettier" plugin

**Comment by paulrberg** [comment]
Ah so that's what you mean by custom rules. Gotcha.
